tests/node-sync/run_light.go

```diff
@@ -51,55 +51,36 @@ func RunLightNode(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 		return err
 	}
 
-	bridgeTotalCh := make(chan int)
-	sub, err := syncclient.Subscribe(ctx, testkit.BridgeTotalTopic, bridgeTotalCh)
-	if err != nil {
-		return err
-	}
-
-	var bridgeTotal int
-	select {
-	case err = <-sub.Done():
-		if err != nil {
-			return err
-		}
-	case bridgeTotal = <-bridgeTotalCh:
-		err = <-syncclient.MustBarrier(ctx, testkit.BridgeStartedState, bridgeTotal).C
-		if err != nil {
-			return err
-		}
-	}
-
-	bridgeCh := make(chan *testkit.BridgeNodeInfo)
-	sub, err = syncclient.Subscribe(ctx, testkit.BridgeNodeTopic, bridgeCh)
+	bridgeCh := make(chan *testkit.BridgeNodeInfo, runenv.IntParam("bridge")) //4
+	sub, err := syncclient.Subscribe(ctx, testkit.BridgeNodeTopic, bridgeCh)
 	if err != nil {
 		return err
 	}
 
 	bridgeNode, err := func(total int) (*testkit.BridgeNodeInfo, error) {
-		for i := 0; i < total; i++ {
+		for {
 			select {
 			case err = <-sub.Done():
 				if err != nil {
 					return nil, err
 				}
+			case <-ctx.Done():
+				return nil,
+					fmt.Errorf("no bridge address has been sent to this light node to connect to")
```

Comment:
Context police 🚓 

The context overhandling detected ⚠️ 

According to this [code](https://github.com/testground/sdk-go/blob/master/sync/client_pubsub.go#L88-L91), the ctx given to Subscribe is already handled there, so you can remove ctx handling here and in other occasions. 

It's a common thing for an object to handle ctx if it rcvs and provides `Done`, which encompasses ctx handling

Reply:
Removed